File: monopoly_data.py
"""Complete American Monopoly game data and rules."""
from typing import List, Dict, Tuple, Optional

# Starting values
STARTING_MONEY = 1500
PASSING_GO_MONEY = 200
LUXURY_TAX = 100
INCOME_TAX = 200
JAIL_POSITION = 10
GO_TO_JAIL_POSITION = 30
JAIL_FINE = 50
MAX_JAIL_TURNS = 3
MAX_HOUSES_PER_PROPERTY = 4

# Property definitions with complete rent information
PROPERTIES: List[Dict] = [
    # Brown Properties
    {
        "name": "Mediterranean Avenue",
        "position": 1,
        "price": 60,
        "rent": [2, 10, 30, 90, 160, 250],  # [base, 1h, 2h, 3h, 4h, hotel]
        "house_cost": 50,
        "mortgage_value": 30,
        "color": (149, 84, 54),
        "group": "Brown"
    },
    {
        "name": "Baltic Avenue",
        "position": 3,
        "price": 60,
        "rent": [4, 20, 60, 180, 320, 450],
        "house_cost": 50,
        "mortgage_value": 30,
        "color": (149, 84, 54),
        "group": "Brown"
    },
    
    # Light Blue Properties
    {
        "name": "Oriental Avenue",
        "position": 6,
        "price": 100,
        "rent": [6, 30, 90, 270, 400, 550],
        "house_cost": 50,
        "mortgage_value": 50,
        "color": (170, 224, 250),
        "group": "Light Blue"
    },
    {
        "name": "Vermont Avenue",
        "position": 8,
        "price": 100,
        "rent": [6, 30, 90, 270, 400, 550],
        "house_cost": 50,
        "mortgage_value": 50,
        "color": (170, 224, 250),
        "group": "Light Blue"
    },
    {
        "name": "Connecticut Avenue",
        "position": 9,
        "price": 120,
        "rent": [8, 40, 100, 300, 450, 600],
        "house_cost": 50,
        "mortgage_value": 60,
        "color": (170, 224, 250),
        "group": "Light Blue"
    },
    
    # Pink Properties
    {
        "name": "St. Charles Place",
        "position": 11,
        "price": 140,
        "rent": [10, 50, 150, 450, 625, 750],
        "house_cost": 100,
        "mortgage_value": 70,
        "color": (217, 58, 150),
        "group": "Pink"
    },
    {
        "name": "States Avenue",
        "position": 13,
        "price": 140,
        "rent": [10, 50, 150, 450, 625, 750],
        "house_cost": 100,
        "mortgage_value": 70,
        "color": (217, 58, 150),
        "group": "Pink"
    },
    {
        "name": "Virginia Avenue",
        "position": 14,
        "price": 160,
        "rent": [12, 60, 180, 500, 700, 900],
        "house_cost": 100,
        "mortgage_value": 80,
        "color": (217, 58, 150),
        "group": "Pink"
    },
    
    # Orange Properties
    {
        "name": "St. James Place",
        "position": 16,
        "price": 180,
        "rent": [14, 70, 200, 550, 750, 950],
        "house_cost": 100,
        "mortgage_value": 90,
        "color": (247, 148, 29),
        "group": "Orange"
    },
    {
        "name": "Tennessee Avenue",
        "position": 18,
        "price": 180,
        "rent": [14, 70, 200, 550, 750, 950],
        "house_cost": 100,
        "mortgage_value": 90,
        "color": (247, 148, 29),
        "group": "Orange"
    },
    {
        "name": "New York Avenue",
        "position": 19,
        "price": 200,
        "rent": [16, 80, 220, 600, 800, 1000],
        "house_cost": 100,
        "mortgage_value": 100,
        "color": (247, 148, 29),
        "group": "Orange"
    },
    
    # Red Properties
    {
        "name": "Kentucky Avenue",
        "position": 21,
        "price": 220,
        "rent": [18, 90, 250, 700, 875, 1050],
        "house_cost": 150,
        "mortgage_value": 110,
        "color": (237, 27, 36),
        "group": "Red"
    },
    {
        "name": "Indiana Avenue",
        "position": 23,
        "price": 220,
        "rent": [18, 90, 250, 700, 875, 1050],
        "house_cost": 150,
        "mortgage_value": 110,
        "color": (237, 27, 36),
        "group": "Red"
    },
    {
        "name": "Illinois Avenue",
        "position": 24,
        "price": 240,
        "rent": [20, 100, 300, 750, 925, 1100],
        "house_cost": 150,
        "mortgage_value": 120,
        "color": (237, 27, 36),
        "group": "Red"
    },
    
    # Yellow Properties
    {
        "name": "Atlantic Avenue",
        "position": 26,
        "price": 260,
        "rent": [22, 110, 330, 800, 975, 1150],
        "house_cost": 150,
        "mortgage_value": 130,
        "color": (254, 242, 0),
        "group": "Yellow"
    },
    {
        "name": "Ventnor Avenue",
        "position": 27,
        "price": 260,
        "rent": [22, 110, 330, 800, 975, 1150],
        "house_cost": 150,
        "mortgage_value": 130,
        "color": (254, 242, 0),
        "group": "Yellow"
    },
    {
        "name": "Marvin Gardens",
        "position": 29,
        "price": 280,
        "rent": [24, 120, 360, 850, 1025, 1200],
        "house_cost": 150,
        "mortgage_value": 140,
        "color": (254, 242, 0),
        "group": "Yellow"
    },
    
    # Green Properties
    {
        "name": "Pacific Avenue",
        "position": 31,
        "price": 300,
        "rent": [26, 130, 390, 900, 1100, 1275],
        "house_cost": 200,
        "mortgage_value": 150,
        "color": (31, 178, 90),
        "group": "Green"
    },
    {
        "name": "North Carolina Avenue",
        "position": 32,
        "price": 300,
        "rent": [26, 130, 390, 900, 1100, 1275],
        "house_cost": 200,
        "mortgage_value": 150,
        "color": (31, 178, 90),
        "group": "Green"
    },
    {
        "name": "Pennsylvania Avenue",
        "position": 34,
        "price": 320,
        "rent": [28, 150, 450, 1000, 1200, 1400],
        "house_cost": 200,
        "mortgage_value": 160,
        "color": (31, 178, 90),
        "group": "Green"
    },
    
    # Dark Blue Properties
    {
        "name": "Park Place",
        "position": 37,
        "price": 350,
        "rent": [35, 175, 500, 1100, 1300, 1500],
        "house_cost": 200,
        "mortgage_value": 175,
        "color": (0, 114, 187),
        "group": "Dark Blue"
    },
    {
        "name": "Boardwalk",
        "position": 39,
        "price": 400,
        "rent": [50, 200, 600, 1400, 1700, 2000],
        "house_cost": 200,
        "mortgage_value": 200,
        "color": (0, 114, 187),
        "group": "Dark Blue"
    },
]

# Railroads
RAILROADS: List[Dict] = [
    {
        "name": "Reading Railroad",
        "position": 5,
        "price": 200,
        "rent": [25, 50, 100, 200],  # Rent for [1, 2, 3, 4] railroads owned
        "mortgage_value": 100,
        "color": (0, 0, 0),
        "group": "Railroad"
    },
    {
        "name": "Pennsylvania Railroad",
        "position": 15,
        "price": 200,
        "rent": [25, 50, 100, 200],
        "mortgage_value": 100,
        "color": (0, 0, 0),
        "group": "Railroad"
    },
    {
        "name": "B. & O. Railroad",
        "position": 25,
        "price": 200,
        "rent": [25, 50, 100, 200],
        "mortgage_value": 100,
        "color": (0, 0, 0),
        "group": "Railroad"
    },
    {
        "name": "Short Line",
        "position": 35,
        "price": 200,
        "rent": [25, 50, 100, 200],
        "mortgage_value": 100,
        "color": (0, 0, 0),
        "group": "Railroad"
    },
]

# Utilities
UTILITIES: List[Dict] = [
    {
        "name": "Electric Company",
        "position": 12,
        "price": 150,
        "mortgage_value": 75,
        "color": (200, 200, 200),
        "group": "Utility"
    },
    {
        "name": "Water Works",
        "position": 28,
        "price": 150,
        "mortgage_value": 75,
        "color": (200, 200, 200),
        "group": "Utility"
    },
]

# Build complete board (40 spaces)
def build_board() -> List[Dict]:
    """Build the complete 40-space Monopoly board."""
    board = [None] * 40
    
    # Special spaces
    board[0] = {"name": "GO", "type": "go"}
    board[2] = {"name": "Community Chest", "type": "community_chest"}
    board[4] = {"name": "Income Tax", "type": "income_tax"}
    board[7] = {"name": "Chance", "type": "chance"}
    board[10] = {"name": "Jail", "type": "jail"}
    board[17] = {"name": "Community Chest", "type": "community_chest"}
    board[20] = {"name": "Free Parking", "type": "free_parking"}
    board[22] = {"name": "Chance", "type": "chance"}
    board[30] = {"name": "Go To Jail", "type": "go_to_jail"}
    board[33] = {"name": "Community Chest", "type": "community_chest"}
    board[36] = {"name": "Chance", "type": "chance"}
    board[38] = {"name": "Luxury Tax", "type": "luxury_tax"}
    
    # Properties
    for prop in PROPERTIES:
        board[prop["position"]] = {**prop, "type": "property"}
    
    # Railroads
    for railroad in RAILROADS:
        board[railroad["position"]] = {**railroad, "type": "railroad"}
    
    # Utilities
    for utility in UTILITIES:
        board[utility["position"]] = {**utility, "type": "utility"}
    
    return board

BOARD_SPACES = build_board()

# Property groups for monopoly checking
PROPERTY_GROUPS = {
    "Brown": [1, 3],
    "Light Blue": [6, 8, 9],
    "Pink": [11, 13, 14],
    "Orange": [16, 18, 19],
    "Red": [21, 23, 24],
    "Yellow": [26, 27, 29],
    "Green": [31, 32, 34],
    "Dark Blue": [37, 39],
    "Railroad": [5, 15, 25, 35],
    "Utility": [12, 28],
}

# Community Chest cards (full list)
COMMUNITY_CHEST_CARDS = [
    {"id": "cc_collect_100", "text": "You set aside time every week to hang out with your elderly neighbor – you've heard some amazing stories! COLLECT $100.", "action": ("money", 100)},
    {"id": "cc_collect_50",  "text": "You organize a group to clean up your town's footpaths. COLLECT $50.", "action": ("money", 50)},
    {"id": "cc_collect_10",  "text": "You volunteered at a blood donation. There were free cookies! COLLECT $10.", "action": ("money", 10)},
    {"id": "cc_pay_50",      "text": "You buy a few bags of cookies from that school bake sale. Yum! PAY $50.", "action": ("money", -50)},
    {"id": "cc_getout",      "text": "GET OUT OF JAIL FREE. Keep this card until needed.", "action": ("jail_free", 1)},
    {"id": "cc_collect_from_each_10", "text": "You organize a street party... COLLECT $10 FROM EACH PLAYER.", "action": ("collect_from_each", 10)},
    {"id": "cc_go_to_jail",  "text": "GO TO JAIL. DO NOT PASS GO.", "action": ("go_to_jail", None)},
    {"id": "cc_collect_20",  "text": "You help your neighbor bring in her groceries. COLLECT $20.", "action": ("money", 20)},
    {"id": "cc_collect_100_b","text": "You help build a new school playground – COLLECT $100.", "action": ("money", 100)},
    {"id": "cc_collect_100_c","text": "You spend the day playing games with kids at a local children's hospital. COLLECT $100.", "action": ("money", 100)},
    {"id": "cc_pay_100",     "text": "You go to the local school's car wash fundraiser – PAY $100.", "action": ("money", -100)},
    {"id": "cc_advance_go",  "text": "ADVANCE TO GO. (COLLECT $200)", "action": ("advance", 0, True)},
    {"id": "cc_collect_200", "text": "You help your neighbors clean up after a storm. COLLECT $200.", "action": ("money", 200)},
    {"id": "cc_pay_50_b",    "text": "Donation to animal shelter. PAY $50.", "action": ("money", -50)},
    {"id": "cc_pay_for_repairs", "text": "For each house pay $40. For each hotel pay $115.", "action": ("pay_per_house_hotel", (40, 115))},
    {"id": "cc_collect_25",  "text": "You organize a bake sale. COLLECT $25.", "action": ("money", 25)}
]

# Chance cards (full list)
CHANCE_CARDS = [
    {"id": "ch_boardwalk", "text": "Advance to Boardwalk.", "action": ("advance", 39, False)},
    {"id": "ch_go",        "text": "Advance to Go (Collect $200).", "action": ("advance", 0, True)},
    {"id": "ch_illinois",  "text": "Advance to Illinois Avenue. If you pass Go, collect $200.", "action": ("advance", 24, True)},
    {"id": "ch_st_charles","text": "Advance to St. Charles Place. If you pass Go, collect $200.", "action": ("advance", 11, True)},
    {"id": "ch_nearest_rail","text": "Advance to the nearest Railroad.", "action": ("advance_nearest", "railroad")},
    {"id": "ch_nearest_rail_b","text": "Advance to the nearest Railroad.", "action": ("advance_nearest", "railroad")},
    {"id": "ch_nearest_utility","text": "Advance token to nearest Utility.", "action": ("advance_nearest", "utility")},
    {"id": "ch_dividend",  "text": "Bank pays you dividend of $50.", "action": ("money", 50)},
    {"id": "ch_getout",    "text": "Get Out of Jail Free.", "action": ("jail_free", 1)},
    {"id": "ch_back_3",    "text": "Go Back 3 Spaces.", "action": ("advance_relative", -3)},
    {"id": "ch_go_to_jail","text": "Go to Jail. Do not pass Go.", "action": ("go_to_jail", None)},
    {"id": "ch_repairs",   "text": "For each house pay $25. For each hotel pay $100.", "action": ("pay_per_house_hotel", (25, 100))},
    {"id": "ch_speeding",  "text": "Speeding fine $15.", "action": ("money", -15)},
    {"id": "ch_reading",   "text": "Advance to Reading Railroad.", "action": ("advance", 5, True)},
    {"id": "ch_chairman",  "text": "Pay each player $50.", "action": ("pay_each_player", 50)},
    {"id": "ch_loan",      "text": "Collect $150.", "action": ("money", 150)}
]

File: ARPi2.Sharp/ARPi2.Sharp/Games/Monopoly/MonopolyData.cs
using System;
using System.Collections.Generic;

namespace ARPi2.Sharp.Games.Monopoly;

// ═══════════════════════════════════════════════════════════════════════════
//  Top-level domain types
// ═══════════════════════════════════════════════════════════════════════════

/// <summary>One of the 40 board positions.</summary>
public class SpaceData
{
    public string Name      { get; init; } = "";
    public string Type      { get; init; } = "";  // go, property, railroad, utility, chance, community_chest, income_tax, luxury_tax, jail, go_to_jail, free_parking
    public int    Position   { get; init; }
    public int    Price      { get; init; }
    public int[]? Rent       { get; init; }         // property: [base,1h,2h,3h,4h,hotel]  railroad: [1,2,3,4 owned]
    public int    HouseCost  { get; init; }
    public int    MortgageValue { get; init; }
    public (int R, int G, int B) Color { get; init; }
    public string? Group     { get; init; }
}

/// <summary>Effect of a Chance / Community Chest card.</summary>
public class CardAction
{
    public string Type { get; init; } = "";      // money, jail_free, go_to_jail, advance, advance_relative, advance_nearest, collect_from_each, pay_each_player, pay_per_house_hotel
    public int    Amount { get; init; }
    public int    Target { get; init; }
    public bool   PassGo { get; init; }
    public string? Nearest { get; init; }         // "railroad" or "utility"
    public int    RepairHouse { get; init; }
    public int    RepairHotel { get; init; }
}

/// <summary>One Chance or Community Chest card.</summary>
public class CardData
{
    public string Id   { get; init; } = "";
    public string Text { get; init; } = "";
    public CardAction Action { get; init; } = new();
}

/// <summary>Complete American Monopoly board data — direct port of data.py.</summary>
public static class MonopolyData
{
    // ─── Constants ─────────────────────────────────────────────────
    public const int StartingMoney       = 1500;
    public const int PassingGoMoney      = 200;
    public const int LuxuryTax           = 100;
    public const int IncomeTax           = 200;
    public const int JailPosition        = 10;
    public const int GoToJailPosition    = 30;
    public const int JailFine            = 50;
    public const int MaxJailTurns        = 3;
    public const int MaxHousesPerProp    = 4;

    // ─── Property groups → board positions ─────────────────────────
    public static readonly Dictionary<string, int[]> PropertyGroups = new()
    {
        ["Brown"]      = [1, 3],
        ["Light Blue"] = [6, 8, 9],
        ["Pink"]       = [11, 13, 14],
        ["Orange"]     = [16, 18, 19],
        ["Red"]        = [21, 23, 24],
        ["Yellow"]     = [26, 27, 29],
        ["Green"]      = [31, 32, 34],
        ["Dark Blue"]  = [37, 39],
        ["Railroad"]   = [5, 15, 25, 35],
        ["Utility"]    = [12, 28],
    };

    // ─── Build the 40-space board ──────────────────────────────────
    public static Dictionary<int, SpaceData> BuildBoard()
    {
        var board = new Dictionary<int, SpaceData>();

        // Special spaces
        board[0]  = new() { Name = "GO",              Type = "go",              Position = 0 };
        board[2]  = new() { Name = "Community Chest",  Type = "community_chest", Position = 2 };
        board[4]  = new() { Name = "Income Tax",       Type = "income_tax",      Position = 4 };
        board[7]  = new() { Name = "Chance",            Type = "chance",          Position = 7 };
        board[10] = new() { Name = "Jail",              Type = "jail",            Position = 10 };
        board[17] = new() { Name = "Community Chest",  Type = "community_chest", Position = 17 };
        board[20] = new() { Name = "Free Parking",     Type = "free_parking",    Position = 20 };
        board[22] = new() { Name = "Chance",            Type = "chance",          Position = 22 };
        board[30] = new() { Name = "Go To Jail",        Type = "go_to_jail",      Position = 30 };
        board[33] = new() { Name = "Community Chest",  Type = "community_chest", Position = 33 };
        board[36] = new() { Name = "Chance",            Type = "chance",          Position = 36 };
        board[38] = new() { Name = "Luxury Tax",        Type = "luxury_tax",      Position = 38 };

        // Properties
        void AddProp(int pos, string name, int price, int[] rent, int hc, int mv, (int,int,int) col, string grp) =>
            board[pos] = new SpaceData { Name=name, Type="property", Position=pos, Price=price, Rent=rent, HouseCost=hc, MortgageValue=mv, Color=col, Group=grp };

        // Brown
        AddProp(1, "Mediterranean Avenue", 60, [2,10,30,90,160,250], 50, 30, (149,84,54), "Brown");
        AddProp(3, "Baltic Avenue",        60, [4,20,60,180,320,450], 50, 30, (149,84,54), "Brown");
        // Light Blue
        AddProp(6, "Oriental Avenue",     100, [6,30,90,270,400,550], 50, 50, (170,224,250), "Light Blue");
        AddProp(8, "Vermont Avenue",      100, [6,30,90,270,400,550], 50, 50, (170,224,250), "Light Blue");
        AddProp(9, "Connecticut Avenue",  120, [8,40,100,300,450,600], 50, 60, (170,224,250), "Light Blue");
        // Pink
        AddProp(11, "St. Charles Place", 140, [10,50,150,450,625,750], 100, 70, (217,58,150), "Pink");
        AddProp(13, "States Avenue",     140, [10,50,150,450,625,750], 100, 70, (217,58,150), "Pink");
        AddProp(14, "Virginia Avenue",   160, [12,60,180,500,700,900], 100, 80, (217,58,150), "Pink");
        // Orange
        AddProp(16, "St. James Place",   180, [14,70,200,550,750,950], 100, 90, (247,148,29), "Orange");
        AddProp(18, "Tennessee Avenue",  180, [14,70,200,550,750,950], 100, 90, (247,148,29), "Orange");
        AddProp(19, "New York Avenue",   200, [16,80,220,600,800,1000], 100, 100, (247,148,29), "Orange");
        // Red
        AddProp(21, "Kentucky Avenue",   220, [18,90,250,700,875,1050], 150, 110, (237,27,36), "Red");
        AddProp(23, "Indiana Avenue",    220, [18,90,250,700,875,1050], 150, 110, (237,27,36), "Red");
        AddProp(24, "Illinois Avenue",   240, [20,100,300,750,925,1100], 150, 120, (237,27,36), "Red");
        // Yellow
        AddProp(26, "Atlantic Avenue",   260, [22,110,330,800,975,1150], 150, 130, (254,242,0), "Yellow");
        AddProp(27, "Ventnor Avenue",    260, [22,110,330,800,975,1150], 150, 130, (254,242,0), "Yellow");
        AddProp(29, "Marvin Gardens",    280, [24,120,360,850,1025,1200], 150, 140, (254,242,0), "Yellow");
        // Green
        AddProp(31, "Pacific Avenue",    300, [26,130,390,900,1100,1275], 200, 150, (31,178,90), "Green");
        AddProp(32, "North Carolina Ave",300, [26,130,390,900,1100,1275], 200, 150, (31,178,90), "Green");
        AddProp(34, "Pennsylvania Ave",  320, [28,150,450,1000,1200,1400], 200, 160, (31,178,90), "Green");
        // Dark Blue
        AddProp(37, "Park Place",        350, [35,175,500,1100,1300,1500], 200, 175, (0,114,187), "Dark Blue");
        AddProp(39, "Boardwalk",         400, [50,200,600,1400,1700,2000], 200, 200, (0,114,187), "Dark Blue");

        // Railroads
        void AddRR(int pos, string name) =>
            board[pos] = new SpaceData { Name=name, Type="railroad", Position=pos, Price=200, Rent=new[]{25,50,100,200}, MortgageValue=100, Color=(0,0,0), Group="Railroad" };
        AddRR(5, "Reading Railroad");
        AddRR(15, "Pennsylvania Railroad");
        AddRR(25, "B. & O. Railroad");
        AddRR(35, "Short Line");

        // Utilities
        void AddUtil(int pos, string name) =>
            board[pos] = new SpaceData { Name=name, Type="utility", Position=pos, Price=150, MortgageValue=75, Color=(200,200,200), Group="Utility" };
        AddUtil(12, "Electric Company");
        AddUtil(28, "Water Works");

        return board;
    }

    // ─── Community Chest cards ─────────────────────────────────────
    public static List<CardData> GetCommunityChestCards() => new()
    {
        new() { Id="cc_collect_100", Text="You set aside time to hang out with your neighbor. COLLECT $100.", Action=new() { Type="money", Amount=100 } },
        new() { Id="cc_collect_50",  Text="You clean up your town's footpaths. COLLECT $50.", Action=new() { Type="money", Amount=50 } },
        new() { Id="cc_collect_10",  Text="You volunteered at a blood donation. COLLECT $10.", Action=new() { Type="money", Amount=10 } },
        new() { Id="cc_pay_50",      Text="You buy cookies from a school bake sale. PAY $50.", Action=new() { Type="money", Amount=-50 } },
        new() { Id="cc_getout",      Text="GET OUT OF JAIL FREE.", Action=new() { Type="jail_free", Amount=1 } },
        new() { Id="cc_collect_10e", Text="You organize a street party. COLLECT $10 FROM EACH.", Action=new() { Type="collect_from_each", Amount=10 } },
        new() { Id="cc_go_to_jail",  Text="GO TO JAIL. DO NOT PASS GO.", Action=new() { Type="go_to_jail" } },
        new() { Id="cc_collect_20",  Text="You help your neighbor. COLLECT $20.", Action=new() { Type="money", Amount=20 } },
        new() { Id="cc_collect_100b",Text="You help build a playground. COLLECT $100.", Action=new() { Type="money", Amount=100 } },
        new() { Id="cc_collect_100c",Text="You play games with kids at hospital. COLLECT $100.", Action=new() { Type="money", Amount=100 } },
        new() { Id="cc_pay_100",     Text="Car wash fundraiser. PAY $100.", Action=new() { Type="money", Amount=-100 } },
        new() { Id="cc_advance_go",  Text="ADVANCE TO GO. (COLLECT $200)", Action=new() { Type="advance", Target=0, PassGo=true } },
        new() { Id="cc_collect_200", Text="You help clean up after a storm. COLLECT $200.", Action=new() { Type="money", Amount=200 } },
        new() { Id="cc_pay_50b",     Text="Donation to animal shelter. PAY $50.", Action=new() { Type="money", Amount=-50 } },
        new() { Id="cc_repairs",     Text="For each house pay $40. For each hotel pay $115.", Action=new() { Type="pay_per_house_hotel", RepairHouse=40, RepairHotel=115 } },
        new() { Id="cc_collect_25",  Text="You organize a bake sale. COLLECT $25.", Action=new() { Type="money", Amount=25 } },
    };

    // ─── Chance cards ──────────────────────────────────────────────
    public static List<CardData> GetChanceCards() => new()
    {
        new() { Id="ch_boardwalk",  Text="Advance to Boardwalk.", Action=new() { Type="advance", Target=39 } },
        new() { Id="ch_go",         Text="Advance to Go (Collect $200).", Action=new() { Type="advance", Target=0, PassGo=true } },
        new() { Id="ch_illinois",   Text="Advance to Illinois Avenue.", Action=new() { Type="advance", Target=24, PassGo=true } },
        new() { Id="ch_st_charles", Text="Advance to St. Charles Place.", Action=new() { Type="advance", Target=11, PassGo=true } },
        new() { Id="ch_near_rr",    Text="Advance to the nearest Railroad.", Action=new() { Type="advance_nearest", Nearest="railroad" } },
        new() { Id="ch_near_rr_b",  Text="Advance to the nearest Railroad.", Action=new() { Type="advance_nearest", Nearest="railroad" } },
        new() { Id="ch_near_util",  Text="Advance to nearest Utility.", Action=new() { Type="advance_nearest", Nearest="utility" } },
        new() { Id="ch_dividend",   Text="Bank pays you dividend of $50.", Action=new() { Type="money", Amount=50 } },
        new() { Id="ch_getout",     Text="Get Out of Jail Free.", Action=new() { Type="jail_free", Amount=1 } },
        new() { Id="ch_back_3",     Text="Go Back 3 Spaces.", Action=new() { Type="advance_relative", Amount=-3 } },
        new() { Id="ch_go_to_jail", Text="Go to Jail. Do not pass Go.", Action=new() { Type="go_to_jail" } },
        new() { Id="ch_repairs",    Text="For each house pay $25. For each hotel pay $100.", Action=new() { Type="pay_per_house_hotel", RepairHouse=25, RepairHotel=100 } },
        new() { Id="ch_speeding",   Text="Speeding fine $15.", Action=new() { Type="money", Amount=-15 } },
        new() { Id="ch_reading",    Text="Advance to Reading Railroad.", Action=new() { Type="advance", Target=5, PassGo=true } },
        new() { Id="ch_chairman",   Text="Pay each player $50.", Action=new() { Type="pay_each_player", Amount=50 } },
        new() { Id="ch_loan",       Text="Collect $150.", Action=new() { Type="money", Amount=150 } },
    };
}
